static/js/apps/eval_retrieval_generation/sxs/app.tsx

```diff
@@ -32,27 +33,57 @@ interface AppPropType {
   sheetIdB: string;
 }
 
-function getSortedQueryIds(docInfos: { a: DocInfo; b: DocInfo }) {
-  const idsA = Object.keys(docInfos?.a?.allQuery || {});
-  const idsB = Object.keys(docInfos?.b?.allQuery || {});
+interface CombinedDocInfo {
+  docInfoA: DocInfo;
+  docInfoB: DocInfo;
+  sortedQueryIds: number[];
+}
+
+function getSortedQueryIds(docInfoA: DocInfo, docInfoB: DocInfo): number[] {
+  const idsA = Object.keys(docInfoA.allQuery || {});
+  const idsB = Object.keys(docInfoB.allQuery || {});
   return idsA
     .filter((id) => idsB.includes(id))
     .map((id) => Number(id))
     .sort((a, b) => a - b);
 }
 
+/** Returns the ID of the first unevaluated query. */
+async function getStartingQueryId(
+  props: AppPropType,
+  sortedQueryIds: number[]
+): Promise<number> {
+  const completedIds = await getRatedQueryIds(
+    props.sheetIdA,
+    props.sheetIdB,
+    props.sessionId
+  );
+  for (const queryId of sortedQueryIds) {
+    if (!completedIds.includes(queryId)) {
+      return queryId;
+    }
+  }
+  // If all evals are complete, show the first query.
+  return sortedQueryIds[0];
+}
+
 export function App(props: AppPropType): JSX.Element {
   const [user, setUser] = useState<User | null>(null);
-  const [docInfos, setDocInfos] = useState<{ a: DocInfo; b: DocInfo }>(null);
+  const [combinedDocInfo, setCombinedDocInfo] = useState<CombinedDocInfo>(null);
   const { setSessionQueryId, sessionQueryId } = useContext(SessionContext);
-  const sortedQueryIds = getSortedQueryIds(docInfos);
-  if (!sessionQueryId && sortedQueryIds.length) {
-    setSessionQueryId(sortedQueryIds[0]);
-  }
+
+  useEffect(() => {
+    if (!sessionQueryId && combinedDocInfo?.sortedQueryIds.length) {
```

Comment:
nit: I don't think we want the `!sessionQueryId` because technically everytime `combinedDocInfo` changes, we would want to getStartingQueryId

Reply:
Oh true! Removed. Also added a cleanup like I did for other useEffects on https://github.com/datacommonsorg/website/pull/4495.